### DISCUSSION TITLE
Getting weird errors while interacting with the snap that's running locally

### DISCUSSION THREAD

**Discussion by kpachhai** [MetaMask/snaps#2216] [General]
Did something change on the recent version of Metamask Flask?

We haven't changed anything major on the snap but it looks like we are getting some weird errors.

We only get these errors while interacting with local version of the snap.

Everything works fine if we connect to the latest version on snaps directory.

For example, 
Here's a basic code to generate a local snap wallet:

```ts
export const generateWallet = async (
  evmAddress: string,
): Promise<HDNodeWallet> => {
  const entropy = await snap.request({
    method: 'snap_getEntropy',
    params: {
      version: 1
    },
  });

  const nodeWallet = HDNodeWallet.fromMnemonic(
    Mnemonic.fromEntropy(entropy),
  ).derivePath(`m/44/3030/0/0/0`);

  return nodeWallet;
};
```

This code is the same in the local snap version and the version that's on snaps directory at snaps.metamask.io

When we interact with the official snap at snaps.metamask.io, everything works fine but if we were to connect to our local version of the snap that's running locally, we get a weird error like the following:

```
[Snap: local:http://localhost:9001] TypeError: cannot derive root path (i.e. path starting with "m/") for a node at non-zero depth 5 (argument="path", value="m/44/3030/0/0/0", code=INVALID_ARGUMENT, version=6.11.1)
```

This is just one of the many errors we are seeing.

**Comment by FrederikBolding** [comment]
Do you have other errors you can share? This particular one doesn't look that weird to me 🤔

**Reply by kpachhai** [reply]
@FrederikBolding the weird part is that nothing has changed in the snap code but the local version of the snap produces this error nonetheless while the official version on snaps directory doesn't
